### ISSUE TITLE
`PhysicalFilesWatcher` and `PhysicalFileProvider` should implement `IDisposable`

### ISSUE THREAD

**Issue by dougbu** [aspnet/FileSystem#132]
`PhysicalFilesWatcher` holds system resources in its `FileSystemWatcher` field. But `PhysicalFilesWatcher` and the (`public`) `PhysicalFileProvider` class containing the watcher do not implement `IDisposable`. System resources are therefore not released until finalization.

**Comment by muratg** [comment]
@dougbu Was this a CR find, or were you investigating some leak?

**Comment by dougbu** [comment]
Found this when investigating file system limits hit (aka `FileSystemWatcher` leaks) doing functional testing w/ DNX Core on Linux. We realized our use of `TestServer` was a mess for other reasons and moved those tests to use an xUnit fixture. That minimized our `TestServer` count enough for testing in this environment. But not all `PhysicalFileProvider` instances are pinned by the server so customers could hit similar issues.

@Tratcher and I discussed the issue back then and I finally got around to filing it.
